lib/open-pgp-key-pair.ts

```diff
@@ -84,22 +83,34 @@ export class PGPSecret extends cdk.Construct implements ICredentialPair {
       handler: 'index.handler',
       timeout: 300,
       runtime: lambda.Runtime.NodeJS810,
-      initialPolicy: [
-        new iam.PolicyStatement()
-          .addActions('secretsmanager:CreateSecret',
-                      'secretsmanager:DeleteSecret',
-                      'secretsmanager:GetSecretValue',
-                      'secretsmanager:UpdateSecret',
-                      'ssm:PutParameter',
-                      'ssm:DeleteParameter')
-          .addAllResources(),
-        new iam.PolicyStatement().addActions(...keyActions).addResource(props.encryptionKey.keyArn)
-      ]
     });
-
-    props.encryptionKey.addToResourcePolicy(
-      new iam.PolicyStatement().addActions(...keyActions).addAllResources().addPrincipal(fn.role!.principal)
-    );
+    fn.addToRolePolicy(new iam.PolicyStatement()
+      .allow()
+      .addActions('secretsmanager:CreateSecret',
+                  'secretsmanager:DeleteSecret',
+                  'secretsmanager:GetSecretValue',
+                  'secretsmanager:UpdateSecret')
+      .addResource(cdk.Stack.find(this).formatArn({
+        service: 'secretsmanager',
+        resource: 'secret',
+        sep: ':',
+        resourceName: `${props.secretName}-??????`
+      })));
+    fn.addToRolePolicy(new iam.PolicyStatement()
```

Comment:
please add some whitespace between the statements

Reply:
Done! Somehow I thought I had those whitespaces... Must have been lost in some of my touchup iterations :)